modules/color-picker/react/spec/ColorInput.spec.tsx

```diff
@@ -1,115 +1,150 @@
+
+        expect(container.querySelector('svg')).toHaveClass('wd-icon-check-small');
+      });
+    });
+
+    describe('with a value and background', () => {
+      test('should render a ColorInput with a value and the value as a background', () => {
+        const {container} = render(<ColorInput value={value} />);
+
+        expect(container.querySelector('div div div[class^="css"]')).toHaveStyle(
+          'background-color: #eee;'
+        );
+      });
+    });
+
+    describe('with disabled attribute', () => {
+      test('should render a disabled ColorInput', () => {
+        const {getByTestId} = render(<ColorInput disabled={true} data-testid={id} />);
+        expect(getByTestId(id)).toBeDisabled();
+      });
+    });
 
-  test('ColorInput should spread extra props', () => {
-    const component = mount(<ColorInput data-propspread="test" />);
-    const input = component
-      .find('input') // TODO: Standardize on prop spread location (see #150)
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
+    describe('with extra, arbitrary props', () => {
+      test('should spread extra props', () => {
+        const attr = 'test';
+        const {getByTestId} = render(<ColorInput data-propspread={attr} data-testid={id} />);
+        expect(getByTestId(id)).toHaveAttribute('data-propspread', attr);
+      });
+    });
+
+    describe('when provided an input ref', () => {
+      test('input ref should be defined', () => {
+        const ref: React.RefObject<HTMLInputElement> = React.createRef();
+        render(<ColorInput inputRef={ref} />);
+        expect(ref.current).toBeDefined();
+      });
+    });
+
+    describe('when provided a value with a hashtag', () => {
+      test('the value is stripped', () => {
+        const {getByTestId} = render(<ColorInput value={'#eee'} data-testid={id} />);
+        expect(getByTestId(id).value).toBe(value);
+      });
+    });
+
+    describe('when provided more than 6 characters as the value', () => {
+      test('the value is stripped', () => {
+        const {getByTestId} = render(<ColorInput value={'123456789'} data-testid={id} />);
+        expect(getByTestId(id).value).toBe('123456');
+      });
+    });
   });
-});
 
-describe('ColorInput Accessibility', () => {
-  test('ColorInput should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <FormField label="Label" inputId="input-plain">
-        <ColorInput />
-      </FormField>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when changed', () => {
+    test('should update value', () => {
+      const {getByTestId} = render(<ColorInput value="" data-testid={id} />);
+
+      expect(getByTestId(id).value).toBe('');
+      getByTestId(id).value = value;
+      expect(getByTestId(id).value).toBe(value);
+    });
+
+    test('should call onChange callback on value change', () => {
```

Comment:
Curious what others think here, I think most of these tests in `'when changed'` could just be functional tests in Cypress. They're not really testing API contracts at this point but rather functionality of the input.

Reply:
Agreed! 👍 I think it's good to check that `onChange` and `onValidColorChange` is called, but a lot of the specific behavioral functionality (hash stripping, too long of input, etc.) could/should be checked in cypress